Makefile

```diff
@@ -1,16 +1,36 @@
-regenerate:
+.PHONY: help regenerate test dependencies build checkers action
+
+# Prefer tools that we've installed
+export PATH := $(HOME)/go/bin:$(PATH)
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+regenerate: ## Re-generate lexers and parsers and pass through goimports
```

Comment:
This looks great 👍 

I do not see the regenerate step in github actions.
The `git diff --exit-code` is also used to check that the newest generated code was checked in.

Otherwise it looks amazing :D

Reply:
Sorry, missed regenerate. :frowning_face: 

One question, when I execute `make regenerate`, the created imports for `errors` and `token` are local paths. I was expecting import with `github.com/jmrtt/gographviz` or better yet with `github.com/awalterschulze/gographviz`. So I'm missing something here.

The import after running `make regenerate` looks like `"home/roquette/gographviz/internal/token"` and `goimport` gives an error as expected. What I'm doing wrong?

Reply:
I saw that is possible to specify the package in `gocc` command-line using `-p`. But it was being ignored since the output is going to `./internal` due to https://github.com/goccmack/gocc/blob/65cbc60dd30507e085bed479695e11aaf6fe9c2e/internal/config/config.go#L184

So, I changed the makefile to pass the package as parameter and generate inside internal folder (`cd internal; gocc -zip -p github.com/awalterschulze/gographviz/internal ../dot.bnf`). Please check if this is the best way to perform this in my last commit.

With this, I updated the generated code and is possible to check using `git diff --exit-code`.

Last run of the action can be found in: https://github.com/jmrtt/gographviz/runs/1205227899

Reply:
Aha I think the problem is the `GOPATH` variable needs to be set.
I think `gocc` uses the `GOPATH` to generate the paths.

Reply:
I think with setting the GOPATH we should be able to get away with the old `gocc -zip -o ./internal/ dot.bnf ` which should then also remove the generated txt files, I hope at least.

Reply:
Yes it seems it uses the `go/build` package to figure out the default package
https://github.com/goccmack/gocc/blob/master/internal/config/config.go#L205

Reply:
Thank you for putting in this much effort.
I think this work is needed and I do appreciate it.

Reply:
Thanks for the all the information! :smile: I will revert my last commit and try setting GOPATH to have `gocc -zip -o ./internal/ dot.bnf` back. I'll let you know when have a new iteration.

Reply:
With GOPATH and correct location, only `token.go` was updated by regenerate. Generated txt files didn't appear.

So, we already have the regeneration step and github actions working correctly (needed to set GOPATH and working directory to have it working).

Last run of github actions with regenerate: https://github.com/jmrtt/gographviz/runs/1205882765

Reply:
Amazing work.

Well done, this looks great.

I think with time of not having a working CI, the token.go got outdated some how.  So including in the pull request makes sense :)